tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,422 @@
+	WaitForCassDcToBeReady(t, namespace)
+}
+
+func DeployClusterWithValues(t *testing.T, namespace, options, customValues string) {
+	log.Printf("Deploying a cluster with %s options using the %s values", options, customValues)
+
+	helmValues := map[string]string{}
+	if options == "default" {
+		helmValues = map[string]string{
+			"reaper.ingress.host": "repair.127.0.0.1.nip.io",
+		}
+	}
+	if options == "minio" {
+		serviceName := MinioServiceName(t)
+		helmValues = map[string]string{
+			"medusa.storage_properties.host": fmt.Sprintf("%s.minio.svc.cluster.local", serviceName),
+		}
+	}
+	deployCluster(t, namespace, customValues, helmValues)
+}
+
+func DeployClusterWithValuesAndHeapSettings(t *testing.T, namespace, options, cassandraHeap, stargateHeap, customValues string) {
+	log.Printf("I can deploy a cluster with %s options, %s Cassandra Heap and %s Stargate heap using the %s values", options, cassandraHeap, stargateHeap, customValues)
+
+	splitOptions := strings.Split(options, "-")
+	medusaEnabled := "true"
+	reaperEnabled := "true"
+	monitoringEnabled := "true"
+
+	if Find(splitOptions, "nomedusa") {
+		medusaEnabled = "false"
+	}
+
+	if Find(splitOptions, "noreaper") {
+		reaperEnabled = "false"
+	}
+
+	if Find(splitOptions, "nomonitoring") {
+		monitoringEnabled = "false"
+	}
+
+	newGenSize, _ := strconv.Atoi(strings.ReplaceAll(strings.ReplaceAll(cassandraHeap, "M", ""), "G", ""))
+	helmValues := map[string]string{
+		"cassandra.heap.size":           cassandraHeap,
+		"cassandra.heap.newGenSize":     strconv.Itoa(newGenSize/2) + "M",
+		"stargate.heapMB":               strings.ReplaceAll(stargateHeap, "M", ""),
+		"medusa.enabled":                medusaEnabled,
+		"reaper.enabled":                reaperEnabled,
+		"reaper-operator.enabled":       reaperEnabled,
+		"kube-prometheus-stack.enabled": monitoringEnabled,
+	}
+	deployCluster(t, namespace, customValues, helmValues)
+}
+
+func WaitForCassDcToBeReady(t *testing.T, namespace string) {
+	cassdcKey := types.NamespacedName{
+		Name:      "dc1",
+		Namespace: namespace,
+	}
+
+	k8sClient, err := CassDcClient()
+	g(t).Expect(err).To(BeNil(), "Couldn't instantiate controller-runtime client with cassdc API")
+
+	g(t).Eventually(func() bool {
+		log.Printf("Checking cassandradatacenter %s state in namespace %s...", cassdcKey.Name, cassdcKey.Namespace)
+		cassdc := &cassdcapi.CassandraDatacenter{}
+		err := k8sClient.Get(context.Background(), cassdcKey, cassdc)
+		if err != nil {
+			t.Logf("Failed getting cassdc: %s", err.Error())
+			return false
+		}
+		return cassdc.Status.CassandraOperatorProgress == cassdcapi.ProgressReady &&
+			cassdc.GetConditionStatus(cassdcapi.DatacenterReady) == v1.ConditionTrue
+	}, retryTimeout, retryInterval).Should(BeTrue())
+}
+
+func resourceWithLabelIsPresent(t *testing.T, namespace, resourceType, label string) bool {
+	switch resourceType {
+	case "pod":
+		pods := getPodsWithLabel(t, namespace, label)
+		if len(pods.Items) == 1 {
```

Comment:
Why not ``return len(pods.Items) == 1`` ?

Reply:
Also, you already have ``CountPodsWithLabel`` method also, so simplifying this would mean ``return CountPodsWithLabel(t, namespace, label) == 1``

Reply:
Good point, I introduced `CountPodsWithLabel` later and didn't remember it could be used here.